solver/src/solver/http_solver/model.rs

```diff
@@ -26,44 +28,56 @@ pub struct OrderModel {
 }
 
 #[derive(Debug, Serialize)]
-pub struct UniswapModel {
-    pub token1: String,
-    pub token2: String,
-    #[serde(with = "serde_with::rust::display_fromstr")]
-    pub balance1: u128,
-    #[serde(with = "serde_with::rust::display_fromstr")]
-    pub balance2: u128,
-    #[serde(with = "serde_with::rust::display_fromstr")]
-    pub fee: f64,
+pub struct PoolModel {
+    #[serde(flatten)]
+    pub parameters: PoolParameters,
     pub cost: CostModel,
     pub mandatory: bool,
 }
 
+#[derive(Debug, Serialize)]
+#[serde(tag = "pool_type")]
+pub enum PoolParameters {
+    ConstantProduct(ConstantProductPoolParameters),
+}
+
+#[derive(Debug, Serialize)]
+pub struct ConstantProductPoolParameters {
+    pub reserves: HashMap<H160, U256Wrapper>,
```

Comment:
I'm not a fan of this wrapper type seeing as we already use `serde_with` and have a `SerializeAs` implementation for `U256`. I suggest using something like this which worked when I tried locally:

```rust
#[serde_as]
#[derive(Debug, Serialize)]
pub struct ConstantProductPoolParameters {
    #[serde_as(as = "HashMap<_, DecimalU256>")]
    pub reserves: HashMap<H160, U256>,
    pub fee: BigDecimal,
}
```

Serializing this to JSON output:
```json
{
  "reserves": {
    "0x0102030405060708091011121314151617181920": "1000000000000000000"
  },
  "fee": "0.003"
}
```

Reply:
I am also happy to introduce this. I will also make a separate PR to remove H160Wrapper in the same way.

Reply:
> I will also make a separate PR to remove H160Wrapper in the same way.

Nice!

Reply:
Nice, I didn't realize this approach worked for hash maps too.

Reply:
For the brave and curious, there is a nice comprehensive guide here: https://docs.rs/serde_with/1.9.4/serde_with/guide/serde_as/index.html